intelmq/bots/parsers/shadowserver/_config.py

```diff
@@ -4131,21 +4131,21 @@ def scan_exchange_identifier(field):
 spam_url = {
     'required_fields': [
         ('time.source', 'timestamp', add_UTC_to_timestamp),
-        ('source.ip', 'ip', validate_ip),
+        ('source.ip', 'src_ip', validate_ip),
     ],
     'optional_fields': [
-        ('source.url', 'url', convert_http_host_and_url, True),
-        ('source.fqdn', 'http_host', validate_fqdn),
-        ('source.asn', 'asn', invalidate_zero),
-        ('source.geolocation.cc', 'geo'),
-        ('source.geolocation.region', 'region'),
-        ('source.geolocation.city', 'city'),
-        ('extra.', 'naics', invalidate_zero),
-        ('extra.', 'sector', validate_to_none),
+        ('destination.url', 'url', convert_http_host_and_url, True),
```

Comment:
The `src_ip` is the IP that originated the spam message and the `ip` is resolved from the URL so I mapped the URL details to _destination_.   Is there a better option?

Reply:
Then the question is, what is this report primarily about? The spam relays or the URLs in the spam mails? Previously it was the URL (mapping `('source.url', 'url', convert_http_host_and_url, True),`). If we change the mapping (`url` to `destination.url` instead of `source.url`) IntelMQ users need to adapt their workflows to get the same result as before.
What do others think, especially users which process Shadowserver feeds?

Reply:
The report is about the URL as the src_ip is secondary.  Using source or destination for the URL isn't ideal.

Reply:
If compatibility is the driving factor, I can map the URL details to _source_.  How should the `src_ip` details be included?

Reply:
Please let me know if this would be preferred:

```
# http://www.shadowserver.org/wiki/pmwiki.php/Services/Spam-URL
spam_url = {
    'required_fields': [
        ('time.source', 'timestamp', add_UTC_to_timestamp),
        ('source.ip', 'ip', validate_ip),
    ],
    'optional_fields': [
        ('source.url', 'url', convert_http_host_and_url, True),
        ('source.fqdn', 'host', validate_fqdn),
        ('extra.relay.asn', 'src_asn', invalidate_zero),
        ('extra.relay.geolocation.cc', 'src_geo', validate_to_none),
        ('extra.relay.geolocation.region', 'src_region', validate_to_none),
        ('extra.relay.geolocation.city', 'src_city', validate_to_none),
        ('extra.relay.naics', 'src_naics', invalidate_zero),
        ('extra.relay.sector', 'src_sector', validate_to_none),
        ('source.asn', 'asn', invalidate_zero),
        ('source.geolocation.cc', 'geo'),
        ('source.geolocation.region', 'region'),
        ('source.geolocation.city', 'city'),
        ('extra.', 'naics', invalidate_zero),
        ('extra.', 'sector', validate_to_none),
        ('extra.', 'source', validate_to_none),
        ('extra.', 'sender', validate_to_none),
        ('extra.', 'subject', validate_to_none),
        ('malware.hash.md5', 'md5', validate_to_none),
    ],
    'constant_fields': {
        'classification.taxonomy': 'abusive-content',
        'classification.type': 'spam',
        'classification.identifier': 'spam-url',
    },
}
```

Reply:
Revised version submitted.

Reply:
I think the `source.ip` (and generally the whole `source` key) should describe the **relay**.

For the information about the spam message (subject, URL and it's IP, etc) I would suggest to use `extra.spam.*` keys.

---

This is something I mentioned in the Intelmq v4 wishlist issue:
https://github.com/certtools/intelmq/issues/2071#issuecomment-1105718722

Having an optional key based on `classification.type` (such as `spam`, `phishing`, etc) to hold additional information. If this is agreed on we can move from `extra.spam.*` keys to just `spam.*` keys.

Reply:
Can two IntelMQ events be generated from one source event?

Reply:
> Can two IntelMQ events be generated from one source event?

Yes. Any parser can generate an arbitrary number of events (also zero!) from one report.

Reply:
I think that will have to wait until support for multiple outputs can be added to the parser.

I have two new reports in the queue to be added next week.  Should I put those in a separate PR?

Reply:
Yes I think it's better to put them in a separate PR.